### PR TITLE
Updated the automatic evaluation code and completed the readme.md file as well as the comments in the previous code

### PR DESCRIPTION
**Pull request by knull-cc**
1. Added data extraction code and automatic evaluation code, including extracting data from raw data and converting it into standard dataset format, as well as using the Bleu metric to evaluate the performance of various original models and fine-tuned models in multi-turn dialogues.
2. Supplemented the tabular data of training data and the result tables of automatic evaluation, all of which are in the corresponding README.md files.
3. Added comprehensive code comments to the previous code, including file comments, code comments, variable comments, etc.

### CODE REVIEW

File: finetune/automatic_evaluation/format_data.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+"""
+This script reads a multi-turn dialogue dataset from a JSON file,
+extracts relevant information, and formats it for further processing.
+"""
+
+import json
+
+# Read raw data
+with open('multi_turn_dataset_1.json', 'r', encoding='utf-8') as f:
```
Comment (weixun2018) [inline finetune/automatic_evaluation/format_data.py]: 修改一下文件名，和上传的json文件一致，后面的output_file也是。

File: finetune/automatic_evaluation/format_data.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    # Only save the first 200 entries
+    if count > 200: 
+        break
+    
+    # Get the complete dialogue rounds and ensure there are at least 3 dialogues
+    if len(conversation) >= 3:  
+        print(f"Dialogue content: {conversation}")  # Print the conversation content
+        
+        # Extract the first three dialogue turns
+        dialogue_history = conversation[:3]  # Extract the first three messages
```
Comment (weixun2018) [inline finetune/automatic_evaluation/format_data.py]: `dialogue_history ` 这个变量是用来干啥的？`auto_evaluation.py`没看到使用这个变量

File: finetune/automatic_evaluation/standard_dataset.json [hunk 3]
```diff
@@ -0,0 +1,4002 @@
...
+            {
+                "input": "谢谢您的鼓励，但我觉得自己学习成绩不好，没有进步，这让我更加困惑。",
+                "output": "学习成绩好坏不是解决问题的关键，专注于自己的进步更为重要。为了达到自己的目标，制定一份合理的计划，按部就班的完成每一项任务。有时候自学不如求助他人，不要害怕问老师和同学问题，以便获得更好的帮助和理解。"
+            },
+            {
+                "input": "我认为我缺乏自信，不知道怎样扭转自己的消极情绪，您能给我一些建议吗？",
+                "output": "建立自信需要时间和努力。让我们从一点一滴的进步中建立自信心。设立一些具体的目标，完成这些目标将会培养自己的自信。鼓励自己，在每天结束前，总结一下自己的成就，这将帮助你更好的看到自身的优点。"
+            }
+        ],
+        "question": "我最近很迷茫，高三后的压力让我感到恐惧，我对自己的存在感到怀疑，您能不能给我一些建议呢？",
```
Comment (weixun2018) [inline finetune/automatic_evaluation/standard_dataset.json]: "question"和"answer"和第一个"input" "output"一样，是啥意思？

File: finetune/README.md [hunk 4]
```diff
@@ -10,4 +10,31 @@
...
+|**MiniCPM-2b**|Base|5'48"|0.052025|0.1473|0.0477|0.0139|0.0032||
+|**MiniCPM-2b-5000**|5,000 samples|5'48"|0.0555|0.1556|0.0483|0.0141|0.0040|✅|
+||||||||||
+|**MiniCPM-2b**|Base|5'48"|0.0573|0.1607|0.0490|0.0152|0.0043|✅|
+|**MiniCPM-2b-10000**|10,000 samples|5'48"|0.0564|0.1554|0.0486|0.0163|0.0053||
+||||||||||
+|**MiniCPM-4b**|Base|18'27"|0.05715|0.1421|0.0585|0.0216|0.0064|✅|
+|**MiniCPM-4b-5000**|5,000 samples|18'27"|0.05405|0.1338|0.0548|0.0207|0.0069||
+||||||||||
+|**MiniCPM3-4B**|Base|12'54"|0.0628|0.1591|0.0622|0.0232|0.0067|✅|
```
Comment (weixun2018) [inline finetune/README.md]: `MiniCPM3-4B`跟`MiniCPM-4b`不一样？
另外，都是Base，得到的结果不一样，是跑了两次Base吗？如果是的话，建议Base只跑一次，每种模型只有三个结果。
还有，这行的运行时间是12'54，上面的是18'27，为啥会有这个差别。（其他又是一样的）

**Review by weixun2018 — COMMENTED** [review]
按照review，再仔细检查一下代码。

File: finetune/automatic_evaluation/format_data.py (hunk 1 above)
Comment (knull-cc) [reply finetune/automatic_evaluation/format_data.py]: 已修改好

File: finetune/automatic_evaluation/format_data.py (hunk 2 above)
Comment (knull-cc) [reply finetune/automatic_evaluation/format_data.py]: dialogue_history  是携带历史消息，只有在 format_data.py 中用到，已经注释到文件中

File: finetune/automatic_evaluation/standard_dataset.json (hunk 3 above)
Comment (knull-cc) [reply finetune/automatic_evaluation/standard_dataset.json]: "input" "output"是直接从原始数据中提取的，input就是user,output就是assistant,question 是 提问内容, answer是 标准回答

File: finetune/README.md (hunk 4 above)
Comment (knull-cc) [reply finetune/README.md]: 对的，每次都跑了一次base，就是base & 对应的微调对比，每次base得分都有微弱差距，其他的都是一个运行时间内跑的（没有重启），最后时间过长可能是重启了一次导致的

File: finetune/automatic_evaluation/format_data.py (hunk 2 above)
Comment (weixun2018) [reply finetune/automatic_evaluation/format_data.py]: 也就是说`dialogue_history`并没有在自动评估里面调用，那生成这个变量是干啥用的？而且在`standard_dataset.json`里面也有。
1. 按照理解，这个变量是历史消息，应该是用来传入模型的，但是`auto_evaluation.py`没有调用，是遗漏了吗？
2. 如果是遗漏了，那`standard_dataset.json`的数据构成让人觉得奇怪，`dialogue_history`第一个记录和question，answer是一样的，这不是重复了吗？

File: finetune/README.md (hunk 4 above)
Comment (weixun2018) [reply finetune/README.md]: 1. MiniCPM3-4B跟MiniCPM-4b是一样的模型，就统一模型名称（其他也是），免得歧义。
2. 同一种模型就是Base，Base-5000，Base-10000三种情况，修改一下表

**Review by weixun2018 — COMMENTED** [review]
还有地方需要修改

File: finetune/automatic_evaluation/auto_evaluation.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            # Construct historical dialogue
+            for turn in dialogue_history:
+                if turn['role'] == 'user':
+                    formatted_dialogue += f"求助者：{turn['content']}\n"
+                elif turn['role'] == 'assistant':
+                    formatted_dialogue += f"支持者：{turn['content']}\n"
+            
+            # Add the current question
+            if formatted_dialogue:
+                formatted_dialogue += "\n历史记录：\n\n"
```
Comment (weixun2018) [inline finetune/automatic_evaluation/auto_evaluation.py]: image

File: finetune/automatic_evaluation/format_data.py
Comment (weixun2018) [inline finetune/automatic_evaluation/format_data.py:1]: 针对多轮对话，我建议这样测试：
假设有3轮，可以构建3条测试样本：
1. 第一个样本以第一轮问答作为输入输出；
2. 第二个样本，以第一轮问答作为history，以第二轮回答作为输入输出；
3. 第三个样本，以第一二轮问答作为history，以第三轮问答作为输入输出。

**Review by weixun2018 — COMMENTED** [review]
有一个bug，还有一个建议。
